.github/workflows/Linter.yml

```diff
@@ -0,0 +1,38 @@
+name: Linter
+on: [pull_request]
+jobs:
+  call-workflow:
+    strategy:
+      matrix:
+        version: ['3.10','3.11','3.12']
```

Comment:
If lint & test are separate tasks (which it looks like they will be here) then I think we don't need to run linters on all 3 python versions, can pick just one.

Sorry, this is my fault I wasn't super precise on the ticket, I was imagining "lint & test" being one job, and that job would then need to be run on all 3 pythons, but what you have in this PR is probably a better approach.